libs/lz4/src/test/java/org/elasticsearch/lz4/ESLZ4CompressorTests.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.lz4;
+
+import net.jpountz.lz4.LZ4Compressor;
+
+import net.jpountz.lz4.LZ4Factory;
+
+import net.jpountz.lz4.LZ4FastDecompressor;
+
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+public class ESLZ4CompressorTests extends ESTestCase {
+
+    public void testCompressRealisticUnicode() {
+        for (int i = 0; i < 15; ++i) {
+            int stringLengthMultiplier = randomFrom(5, 10, 20, 40, 80, 160, 320);
+
+            final String uncompressedString = randomRealisticUnicodeOfCodepointLength(stringLengthMultiplier * 1024);
+            byte[] uncompressed = uncompressedString.getBytes(StandardCharsets.UTF_8);
+
+            byte[] compressed = new byte[uncompressed.length + uncompressed.length / 255 + 16];
+            byte[] unForkedCompressed = new byte[uncompressed.length + uncompressed.length / 255 + 16];
+            LZ4Compressor compressor = ESLZ4Compressor.INSTANCE;
```

Comment:
I wonder if we can validate that on java 9+, we get our own variant?

Reply:
We already run tests with various runtimes. Perhaps we could add a unit test with an `assume` condition based on Java runtime version that verifies the type returned here?